### PR TITLE
Fix CI and example

### PR DESCRIPTION
**Pull request by milesfrain**
Add missing file for CI

### CODE REVIEW

**Comment by garyb** [comment]
Maybe I've not been keeping up with what we're doing with these, but adding `package-lock` has never been useful on PS projects IME (start getting spammed by dependabot/security warnings that wouldn't otherwise arise), I add it to gitignore instead usually.

I guess maybe it's a bit different since this one has actual dependencies, where as usually they're dev, but 🤷, just my 2p.

**Comment by thomashoneyman** [comment]
@garyb they’re only being added for projects which use JS dependencies (like this one). Also, as we’ve experienced on other repositories, removing the lock file unfortunately doesn’t appear to remove the dependabot spam; you still get it just based on the package.json file alone.

By the way — are you aware of a reason _not_ to switch to strict equality in these underlying JS files? They’re failing the eslint check, which is the same config file used with Aff.

That said — to reduce file clutter we can also remove the lock files as they don’t do anything besides speed up CI for the NPM installation.

**Comment by garyb** [comment]
I just meant "more" dependabot spam :smile: as it will warn you about things that the `^` ranges would skip when there are fixed version in the lock that are insecure, I think? I'm probably wrong. But yeah, I realised while writing the above that this does have a real dependency, so if that's the way we're dealing with it, I'm okay with it :+1:

Switching to strict equality sounds good to me too, probably just an oversight that it's not already.

**Comment by thomashoneyman** [comment]
I think it's acceptable to not commit lockfiles and only commit the package.json files. I will need to update the template files for CI as the `npm ci` command won't work and we'll need to use ordinary `npm install`, and template files for .gitignore to add the package-lock.json file.

That's because we aren't actually providing this dependency to users via this library. They have to actually install it themselves. We'll still receive security alerts, but as you said, not ones that would already be covered by the `^` ranges.

With that in mind, @milesfrain, this PR will have to be updated to remove the lock file again and update the .gitignore. But the eslint issues do still need to be resolved. You might want to wait for those updates in the governance repository before doing anything else.

**Comment by milesfrain** [comment]
> You might want to wait for those updates in the governance repository before doing anything else.

Sounds good. I have some other fish to fry over this next week, so no rush.

**Comment by thomashoneyman** [comment]
@milesfrain I've updated the .gitignore and ci.yml files and removed the package-lock.json file. You should be good to go to fix the `eslint` errors, after which this will be all done!

File: example/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+```
+spago bundle-app -m Example.Main -t example/example.js
+npm i -g parcel
+parcel serve example/index.html --open
+```
```
Comment (milesfrain) [inline example/README.md:5]: Kinda terse, but better than nothing.

Comment (thomashoneyman) [reply example/README.md:5]: Is parcel needed here? Can the file just be opened in the browser directly?

**Review by thomashoneyman — APPROVED** [review]

File: example/README.md (hunk 1 above)
Comment (milesfrain) [reply example/README.md:5]: It's not working for me without the bundler. This section is likely the issue:
https://github.com/purescript-contrib/purescript-ace/blob/23355ca6ca0ef9b85ca539330eaafdb6c1c49cfa/example/index.html#L19-L26
The path looks correct to me. Not sure if there's a non-bundler fix.

Comment (thomashoneyman) [reply example/README.md:5]: Ok. If there’s a way to skip the bundled that would be ideal, but this is already a big improvement.

File: example/src/Main.purs [hunk 2]
```diff
@@ -77,15 +77,15 @@ main = onLoad $ do
...
+    >>= KeyBinding.addKeyboardHandler \{editor : editor_shadow} hs kstring kcode evt -> do
       if hs == -1 || (kcode <= 40 && kcode >= 37)
         then pure Nothing
         else do
         Ace.Position {row: startRow, column: startColumn}
           <- Anchor.getPosition startAnchor
         Ace.Position {row: endRow, column: endColumn}
           <- Anchor.getPosition endAnchor
-        selectedRange <- Editor.getSelectionRange editor
+        selectedRange <- Editor.getSelectionRange editor_shadow
```
Comment (thomashoneyman) [inline example/src/Main.purs]: Just to quickly check — what’s the reason for this change? Was it not working before?

Comment (milesfrain) [reply example/src/Main.purs]: To avoid a shadowed variable error / warning. Another naming option is `editor'`

Comment (thomashoneyman) [reply example/src/Main.purs]: Oh, interesting, I didn’t realize this was shadowed before. Let’s do with editor’ and I can merge.

**Pull request merged**